### ISSUE TITLE
Tree navigation does not work

### ISSUE THREAD

**Issue by kryshac** [VSCodeVim/Vim#3466]
**Describe the bug**
If we set "workbench.list.keyboardNavigation" with "filter" or "highlight" does not work as long as the vim extension is active

**Screenshots**
without the vim extension: 
![image](https://user-images.githubusercontent.com/19764133/52398562-27b97b80-2ac2-11e9-9054-d5f4840886b1.png)

with the vim extension:
![image](https://user-images.githubusercontent.com/19764133/52398624-60f1eb80-2ac2-11e9-9d3e-7986dd6fb84c.png)

**Comment by Tomotoes** [comment]
I'm having the same problem.

**Comment by asilvadesigns** [comment]
same. I thought this PR was supposed to fix this https://github.com/VSCodeVim/Vim/pull/3432

**Comment by RononDex** [comment]
Having the same issue here

**Comment by jpoon** [comment]
@joaomoreno

**Comment by joaomoreno** [comment]
The VIM extension defines custom keybindings for tree navigation, as always: `hjkl`, for example.

Because of that, you must toggle the tree filter mode. VIM configures <kbd>/</kbd> for that. So, you press <kbd>/</kbd> and then you type.

**Comment by BosEriko** [comment]
I get this as well but random. When I close and open my VS Code it sometimes works normally for me.

**Comment by asilvadesigns** [comment]
@joaomoreno @jpoon documentation would be great, I'm still not sure if I have a problem with my setup or it just isn't working. Thanks for all the harrd work!

**Comment by coffenbacher** [comment]
How do I escape `/` filter mode and get back to `hjkl`? Generally I `/` for a file and find it, but then can't get back to regular browsing. Is `esc` the only option? That's okay, just a bit distant for me...

**Comment by jpoon** [comment]
Feel free to submit a PR updating our readme (maybe the FAQ section?). I am trying my hardest to adhere to https://twitter.com/jasonthepoon/status/1093375866929786880 :)

**Comment by damassi** [comment]
I've found that search works and then stops working, same with navigation and file / folder deletion. All in all I think the new Explorer (plus VSCode-Vim) has quite a few bugs to iron out before its stable.

**Comment by fengchaoyuan** [comment]
A problem for me is that after using toggleVim command, the explorer sidebar still uses the vim hjkl movement shortcuts and not the highlight search. Only after manually disabling the whole extension could I get the highlight search on without needing to use the / key.

**Comment by jpoon** [comment]
> How do I escape `/` filter mode and get back to `hjkl`? Generally I `/` for a file and find it, but then can't get back to regular browsing. Is `esc` the only option? That's okay, just a bit distant for me...

@coffenbacher as far as i can tell, it's just `esc` for now. 

> A problem for me is that after using toggleVim command, the explorer sidebar still uses the vim hjkl movement shortcuts and not the highlight search. Only after manually disabling the whole extension could I get the highlight search on without needing to use the / key.

@fengchaoyuan Yes, that is by-design. We can't disable this sort of functionality with `toggleVim` as it's all defined in the package.json (ie. not in the js code). I've actually been wanting to remove `toggleVim`, as it's a bit confusing and the implementation is incomplete, but wasn't sure if people used it. Did you have any other intentions with using it?

**Comment by jon301** [comment]
@jpoon actually I use toggleVim a lot when doing pair programming with other folks who don't use Vim
Please don't remove it it's quite useful :)

**Comment by RononDex** [comment]
I found that the issue with vim is that only shift+7 works in the explorer to activate the search.

Note that I am on a swiss-german keyboard, which has a slash on the numpad and one on shift+7.
Even with vim disabled, only shift+7 works for me

**Comment by quittenkaes** [comment]
On my German keyboard I have to hit the "#" key to activate the search. "Shift+7" would have been correct ("/"), but that doesn't work on the tree, only in the editor.

**Comment by wrightwriter** [comment]
> I've found that search works and then stops working, same with navigation and file / folder deletion. All in all I think the new Explorer (plus VSCode-Vim) has quite a few bugs to iron out before its stable. (I ended up downgrading to November release.)

You don't have to downgrade. Just switch workbench.list.keyboardNavigation to "simple" in settings.

**Comment by asilvadesigns** [comment]
I can't get this to work with any setting. When `workbench.list.keyboardNavigation` set to `simple`, the first letter I type jumps to an item of that letter... not desirable at all, if I set it to `filter` or `highlight` anything I type is treated at a filter or highlight. I've read some people say typing `/` is supposed to toggle this functionality, it's not working for me, neither can I use `hjkl` to navigate the explorer.

**Comment by damassi** [comment]
> workbench.list.keyboardNavigation to "simple" in settings

In my experience this is still super buggy.

**Comment by wrightwriter** [comment]
workbench.list.keyboardNavigation "simple" is working perfectly for me. I have been using it for two days work hjkl navigation.

**Comment by evenfrost** [comment]
Same for me as for @asilvadesigns. It is said that only pressing `/` should toggle new quick search mode but for me it's always on. I'm using `hjkl` navigation in file explorer a lot, but now I cannot and this is rather frustrating.

**Comment by evenfrost** [comment]
UPD: Actually it does work until I switch to Search sidebar. When I go back to File Explorer, navigation with `hjkl` stops working, and I have to reload window to get it back to normal.

**Comment by joaomoreno** [comment]
@asilvadesigns @evenfrost What versions of Code and Vim extension do you have installed?

**Comment by asilvadesigns** [comment]
@joaomoreno thanks for looking into this:
VS Code: `Version 1.31.1 (1.31.1)`
Vim Extension: `1.0.8`

**Comment by evenfrost** [comment]
@joaomoreno Same for me.

**Comment by joaomoreno** [comment]
I could repro that after a while, the toggle mode stops working! :thinking: Fix coming!

**Comment by joaomoreno** [comment]
Pushed fix on VS Code side. Unfortunately you're going to have to wait for the next Stable or simply get tomorrow's Insiders!

**Comment by leoparis89** [comment]
There remains a bug: pressing `L` doesn't open folders in VsCode Version 1.32.1 
Is this fixable via some configuration ?

**Comment by joaomoreno** [comment]
@leoparis89 What VIM extension version do you have installed?

**Comment by BosEriko** [comment]
I'm having the same issue as @leoparis89 right now. Pressing `J` closes the folder but `L` doesn't open folders.

Also, there are times that the VIM navigation on explorer does not work at all. Only fix I have for it is to restart VSCode itself.

My VIM extension version is 1.1.0. MY VSCode version is 1.32.2.

**Comment by joaomoreno** [comment]
> L doesn't open folders.

Yup, that seems to be a new bug: https://github.com/Microsoft/vscode/issues/70681

> Also, there are times that the VIM navigation on explorer does not work at all. Only fix I have for it is to restart VSCode itself.

Can you try to reproduce this situation?

**Comment by leoparis89** [comment]
@joaomoreno 

I am running
Extension (VsCodeVim) version: 1.1.0
VSCode version: Version 1.32.1 (1.32.1)
OS: mac OS 10.13.6

I created issue [#3576](https://github.com/VSCodeVim/Vim/issues/3576)

**Comment by joaomoreno** [comment]
@leoparis89 This is: https://github.com/Microsoft/vscode/issues/70681

**Comment by theholy7** [comment]
@joaomoreno still having this problem. Can you leave a small update where this issue is at?

Currently on `Version: 1.60.1 (Universal)` and v1.21.7.

**Comment by shinichy** [comment]
Same issue for me, too.

Extension (VsCodeVim) version: 1.21.7
VSCode version: Version: 1.60.2 (Universal)
OS: Darwin x64 20.6.0

**Comment by devadathanmb** [comment]
Do we have a solution for this yet?
